qcodes/dataset/measurements.py

```diff
@@ -82,7 +84,9 @@ def __init__(
         dataset: DataSetProtocol,
         write_period: float,
         interdeps: InterDependencies_,
+        span: trace.Span,
```

Comment:
not sure how relevant it is, but perhaps it should be possible to create a datasaver without a span (e.g. with None) for e.g. unit tests? otherwise a span needs to be created always if one wants to create a datasaver?

Reply:
but yes there is no reason not to allow this to be none too